src/core_read.cpp

```diff
@@ -126,31 +126,53 @@ static bool CheckTxScriptsSanity(const CMutableTransaction& tx)
 
 static bool DecodeTx(CMutableTransaction& tx, const std::vector<unsigned char>& tx_data, bool try_no_witness, bool try_witness)
 {
+    CMutableTransaction tx_wit, tx_nowit;
+    bool ok_wit = false, ok_nowit = false;
```

Comment:
naming nit: not sure if this is better, but maybe `wtx`, `tx`, `wtx_ok`, `tx_ok`

Reply:
I've renamed everything to {tx,ok}_{extended,legacy}, because extended serialization doesn't imply the result actually is a segwit transaction - it just means prefer the interpretation of extended serialization when there is ambiguity.